infrastructure/application_stack.py

```diff
@@ -33,8 +33,20 @@ def __init__(self, scope: constructs.Construct, stack_id: str) -> None:
                 conditions={"StringEquals": {"SAML:aud": "https://signin.aws.amazon.com/saml"}},
             )
         else:
-            principal = aws_iam.AccountPrincipal(
-                account_id=aws_iam.AccountRootPrincipal().account_id
+            open_id_connect_provider_arn = (
+                f"arn:aws:iam::"
+                f"{aws_iam.AccountRootPrincipal().account_id}"
+                f":oidc-provider/token.actions.githubusercontent.com"
+            )
+
+            principal = aws_iam.WebIdentityPrincipal(
```

Comment:
You might need a `CompositePrincipal` which also includes the original account root principal here to fix the S3 role assumption.

Reply:
Commit afbc25df6d00a2e9861c6fb80730a30258d18153 seems to have indeed fixed this issue.